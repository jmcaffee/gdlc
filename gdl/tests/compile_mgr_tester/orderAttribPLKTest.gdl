/* **************************************************************************
 *  File: powerLookupTest.gdl
 *
 *  Description:
 *    These are guideline definitions for powerlookup testing.
 *
 ************************************************************************* */

dpm numeric var1;
dpm numeric var2;
dpm numeric var3;
dpm numeric var4;
dpm text    result1;
dpm text    result2;

import( powerlookup, "orderattribplk.csv" );    // Import the powerlookup CSV file



guideline ("OrderAttribPLKTest")
  ruleset   OrderAttribPLK();
end

