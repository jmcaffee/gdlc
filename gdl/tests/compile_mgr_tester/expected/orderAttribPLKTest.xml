<GuidelineRoot><Guideline GuidelineID="1" Name="OrderAttribPLKTest" Version="1" StartDate="May 07 2013 12:00AM"><Ruleset Id="1" Name="Order Attrib PLK" ExecuteType="3" Type="1"><Rule Id="1" Name="Order Attrib PLK.1"><IfMessages><Message Type="Exceptions"><![CDATA[Message 1.]]></Message></IfMessages><ElseMessages><Message Type="Exceptions"><![CDATA[Message 2.]]></Message></ElseMessages><Condition><Expression><Compute><Operator>AND</Operator><LeftOperand><Compute><Operator>AND</Operator><LeftOperand><Compute><Operator>AND</Operator><LeftOperand><Brace><Compute><Operator>==</Operator><LeftOperand><DPM Name="var1"/></LeftOperand><RightOperand><Constant>1</Constant></RightOperand></Compute></Brace></LeftOperand><RightOperand><Brace><Compute><Operator>&lt;&gt;</Operator><LeftOperand><DPM Name="var2"/></LeftOperand><RightOperand><Constant>2</Constant></RightOperand></Compute></Brace></RightOperand></Compute></LeftOperand><RightOperand><Brace><Compute><Operator>==</Operator><LeftOperand><DPM Name="var3"/></LeftOperand><RightOperand><Constant>3</Constant></RightOperand></Compute></Brace></RightOperand></Compute></LeftOperand><RightOperand><Brace><Compute><Operator>&lt;&gt;</Operator><LeftOperand><DPM Name="var4"/></LeftOperand><RightOperand><Constant>4</Constant></RightOperand></Compute></Brace></RightOperand></Compute></Expression></Condition><IfActions><Assign><AssignTo><DPM Name="result1"/></AssignTo><AssignValue><Expression><Constant>Fail</Constant></Expression></AssignValue></Assign><Assign><AssignTo><DPM Name="result2"/></AssignTo><AssignValue><Expression><Constant>Y</Constant></Expression></AssignValue></Assign></IfActions></Rule><Rule Id="2" Name="Order Attrib PLK.2"><IfMessages><Message Type="Exceptions"><![CDATA[Message 3.]]></Message></IfMessages><ElseMessages><Message Type="Exceptions"><![CDATA[Message 4.]]></Message></ElseMessages><Condition><Expression><Compute><Operator>AND</Operator><LeftOperand><Compute><Operator>AND</Operator><LeftOperand><Compute><Operator>AND</Operator><LeftOperand><Brace><Compute><Operator>==</Operator><LeftOperand><DPM Name="var1"/></LeftOperand><RightOperand><Constant>5</Constant></RightOperand></Compute></Brace></LeftOperand><RightOperand><Brace><Compute><Operator>&lt;&gt;</Operator><LeftOperand><DPM Name="var2"/></LeftOperand><RightOperand><Constant>6</Constant></RightOperand></Compute></Brace></RightOperand></Compute></LeftOperand><RightOperand><Brace><Compute><Operator>==</Operator><LeftOperand><DPM Name="var3"/></LeftOperand><RightOperand><Constant>7</Constant></RightOperand></Compute></Brace></RightOperand></Compute></LeftOperand><RightOperand><Brace><Compute><Operator>&lt;&gt;</Operator><LeftOperand><DPM Name="var4"/></LeftOperand><RightOperand><Constant>8</Constant></RightOperand></Compute></Brace></RightOperand></Compute></Expression></Condition><IfActions><Assign><AssignTo><DPM Name="result1"/></AssignTo><AssignValue><Expression><Constant>Fail</Constant></Expression></AssignValue></Assign><Assign><AssignTo><DPM Name="result2"/></AssignTo><AssignValue><Expression><Constant>N</Constant></Expression></AssignValue></Assign></IfActions></Rule></Ruleset></Guideline><LOOKUPS></LOOKUPS><Conditions></Conditions><DERIVEDPARAMETERS><DPM Name="result1" Type="DPM" Order="0" ProductType="6" DataType="Text"/><DPM Name="result2" Type="DPM" Order="0" ProductType="6" DataType="Text"/><DPM Name="var3" Type="DPM" Order="0" ProductType="4" DataType="Numeric"/><DPM Name="var4" Type="DPM" Order="0" ProductType="4" DataType="Numeric"/><DPM Name="var1" Type="DPM" Order="0" ProductType="4" DataType="Numeric"/><DPM Name="var2" Type="DPM" Order="0" ProductType="4" DataType="Numeric"/></DERIVEDPARAMETERS></GuidelineRoot>